contracts/VeListaAutoCompounder.sol

```diff
@@ -0,0 +1,330 @@
+    // 300 by default (3%)
+    uint256 public feeRate;
+
+    // The maximum fee rate (bps) for auto-compounding
+    // 10% by default
+    uint256 public maxFeeRate;
+
+    // The maximum fee for auto-compounding
+    // $100M by default
+    uint256 public maxFee;
+
+    // The minimum USD value required to lock for auto compounding
+    // $5 by default
+    uint256 public autoCompoundThreshold;
+
+    // user => auto compound status (default / enabled / disabled)
+    mapping(address => Status) private autoCompoundStatus;
+
+    // Flag to indicate the meaning of Status.Default;
+    // If true, auto-compound is enabled for users whose status is `Default`
+    // If false, auto-compound is disabled for users whose status is `Default`
+    // Set to true initially
+    bool public enableByDefault;
+
+    // The total auto-compounding fee collected (in LISTA)
+    uint256 public totalFee;
+
+    bytes32 public constant BOT = keccak256("BOT");
+
+    enum Status {
+        Default,
+        Enabled,
+        Disabled
+    }
+
+    /********************** Events ***********************/
+    event VeListaDistributorUpdated(address indexed _veListaDistributor);
+    event OracleUpdated(address indexed _oracle);
+    event AutoCompoundStatus(address indexed _account, bool _enabled);
+    event AutoCompounded(
+        address indexed _account,
+        address _lista,
+        uint256 _claimedAmount,
+        uint256 _amtToCompound
+    );
+    event FeeWithdrawn(address indexed _receiver, uint256 _fee);
+    event FeeReceiverUpdated(address indexed _newReceiver);
+
+    function initialize(
+        address _lista,
+        address _velista,
+        address _veListaDistributor,
+        address _oracle,
+        address _feeReceiver,
+        address _admin,
+        address _bot
+    ) public initializer {
+        require(
+            _lista != address(0) &&
+                _velista != address(0) &&
+                _veListaDistributor != address(0) &&
+                _feeReceiver != address(0) &&
+                _admin != address(0) &&
+                _bot != address(0) &&
+                _oracle != address(0),
+            "Zero address provided"
+        );
+        __AccessControl_init();
+        _setupRole(DEFAULT_ADMIN_ROLE, _admin);
+        _setupRole(BOT, _bot);
+
+        lista = IERC20(_lista);
+        veLista = IVeLista(_velista);
+        require(lista == veLista.token(), "Invalid Lista token");
+
+        veListaDistributor = IVeListaDistributor(_veListaDistributor);
+        require(veLista == veListaDistributor.veLista(), "Invalid veLista");
+
+        oracle = OracleInterface(_oracle);
+        feeReceiver = _feeReceiver;
+
+        feeRate = 300; // 3%
+        maxFeeRate = 1000; // 10%
+        require(
+            feeRate <= maxFeeRate && maxFeeRate <= 10000,
+            "Invalid fee rate"
+        );
+
+        autoCompoundThreshold = 5 * 1e18; // $5
+        maxFee = 1000000 * 1e18; // $100M
```

Comment:
this comment is wrong

Reply:
thanks. done in 9bf045eb0ebca5bca16b467b3bc4f99426824a84